client/packages/system/src/Item/DetailView/DetailView.tsx

```diff
@@ -25,15 +25,15 @@ export const ItemDetailView: FC = () => {
     setSuffix(data?.name ?? '');
   }, [data]);
 
-  if (isLoading) return <DetailFormSkeleton />;
+  if (isLoading || !data) return <DetailFormSkeleton />;
```

Comment:
interesting that this is needed - did you have a situation where it came up?
My concern is that when there's an error and that results in `!data` then the skeleton is perpetually shown, when we should be showing some specific error. I think though, don't change - this is better than the page crashing

Reply:
We could show a specific error instead. Was this set up already, or should I create one with the standard icons etc? 

Andrei brought how he didn't like uncertain types in one of the multi unit PRs, so I've just brought this over here

Reply:
It does bring in a change to functionality, though it's not something I expect will happen.

There's already a check for `!!data` below (line 41) which will handle the 'no data' situation. 
The change had me wondering if there's a need for a wrapper component to handle, though maybe this is a bit clumsy, adding in the messaging and routing?

```
interface DataViewWrapperProps {
  isLoading: boolean;
  isNoData: boolean;
  noDataTitle: string;
  noDataMessage: string;
  onOk: () => void;
  skeleton: React.ReactNode;
}

export const DataViewWrapper: React.FC<
  PropsWithChildren<DataViewWrapperProps>
> = ({
  children,
  isLoading,
  isNoData,
  noDataMessage,
  noDataTitle,
  onOk,
  skeleton,
}) => {
  if (isLoading) return skeleton;

  if (isNoData)
    return (
      <AlertModal
        open={true}
        onOk={onOk}
        title={noDataTitle}
        message={noDataMessage}
      />
    );

  return children;
};
```

and then this component becomes

```
export const ItemDetailView: FC = () => {
  const { data, isLoading } = useItem();
  const navigate = useNavigate();
  const t = useTranslation('catalogue');
  const { setSuffix } = useBreadcrumbs();

  React.useEffect(() => {
    setSuffix(data?.name ?? '');
  }, [data]);

  const tabs = [
    {
      Component: <GeneralTab />,
      value: 'General',
    },
    {
      Component: <MasterListsTab />,
      value: 'Master Lists',
    },
  ];

  return (
    <DataViewWrapper
      skeleton={<DetailFormSkeleton />}
      isLoading={isLoading}
      isNoData={!data}
      noDataTitle={t('error.item-not-found')}
      noDataMessage={t('messages.click-to-return-to-item-list')}
      onOk={() =>
        navigate(
          RouteBuilder.create(AppRoute.Catalogue)
            .addPart(AppRoute.Items)
            .build()
        )
      }
    >
      <Box style={{ width: '100%' }}>
        <Toolbar />
        <DetailTabs tabs={tabs} />
      </Box>
    </DataViewWrapper>
  );
};
```

Reply:
> It does bring in a change to functionality, though it's not something I expect will happen.
> 
> There's already a check for `!!data` below (line 41) which will handle the 'no data' situation. The change had me wondering if there's a need for a wrapper component to handle, though maybe this is a bit clumsy, adding in the messaging and routing?

Can remove the check above for now and contemplate a no data component later 🤔